django_graph_api/tests/graphql/test_types.py

```diff
@@ -0,0 +1,47 @@
+import pytest
+
+from unittest import mock
+
+from django_graph_api.graphql.schema import Schema
+from django_graph_api.graphql.types import Boolean, Float, Field, Int, String
+
+schema = Schema()
+
+
+def test_field_get_value_calls_coerce():
+    field = Field()
+    field.type_ = mock.Mock()
+    field.name = 'foo'
+    field.obj = mock.Mock()
+    field.obj.get_foo.return_value = 'bar'
+
+    field.get_value()
+    field.type_.coerce_result.assert_called_once_with('bar')
+
+
+def test_boolean_coerce():
+    assert Boolean.coerce_result(1) is True
+    assert Boolean.coerce_result(None) is None
+
+
+def test_int_coerce():
+    assert Int.coerce_result(True) is 1
+    assert Int.coerce_result(4.9) is 4
+    assert Int.coerce_result(None) is None
+
+    with pytest.raises(ValueError):
+        Int.coerce_result('2.0')
+
+
+def test_float_coerce():
+    assert Float.coerce_result(-10) == -10.0
+    assert Float.coerce_result(None) is None
+
+    with pytest.raises(ValueError):
+        Float.coerce_result('abc')
+
+
+def test_string_coerce():
+    assert String.coerce_result(True) == 'True'
```

Comment:
What do you think about these test cases? I didn't want to test too much because at that point it's just testing the built-in casting. But if you think more test cases would be good for like documentation purposes, then I can add more.

Reply:
Also, btw, pytest has [parameterized tests](https://docs.pytest.org/en/latest/parametrize.html#pytest-mark-parametrize-parametrizing-test-functions) which is kind of cool, but is a little awkward to read. `unittest.subTest` doesn't work with pytest :(

Reply:
re: the test cases, these look good to me. I do think it's worth having the tests even though it's the built-in behavior of python because what we're testing is that the behavior is what's expected by the spec. it's possible the coerce_result method could be implemented in a different way in the future that doesn't rely on that built-in behavior, because it can't for some reason.

Tbh looking back over the spec it is actually pretty sparse regarding what values are valid for result coercion and what values aren't. :-p Input coercion looks a little more well-defined / has more stringent requirements. The amount of tests you have here is probably sufficient for now.

Reply:
Re: parameterized tests, neat! Reminds me a bit of the [hypothesis](https://hypothesis.readthedocs.io/) that was mentioned in the testing talk...

Reply:
Oh yeah, speaking of that talk, I need to remember to try out [mutpy](https://pypi.python.org/pypi/MutPy/0.4.0) on this codebase at some point.

Reply:
YES I AM SO EXCITED TO TRY THAT ALSO